src/Primitive/Stringg.php

```diff
@@ -0,0 +1,62 @@
+<?php
+namespace Widmogrod\Primitive;
+
+use Widmogrod\Common;
+use Widmogrod\FantasyLand;
+
+class Stringg implements FantasyLand\Primitive
+{
+    const of = 'Widmogrod\Primitive\PString::of';
+
+    use Common\PointedTrait;
+    use Common\ValueOfTrait;
+
+    /**
+     * @inheritdoc
+     */
+    public static function mempty()
+    {
+        return self::of("");
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getEmpty()
+    {
+        return self::mempty();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function concat(FantasyLand\Semigroup $value)
+    {
+        if ($value instanceof self) {
+            return $value->map(function ($b) {
```

Comment:
Why use `map` since we know that both objects are of the same type we can safely access the inner value directly avoiding functions call overhear ?